modin/pandas/dataframe.py

```diff
@@ -394,12 +394,14 @@ def apply(
         result_type=None,
         args=(),
         by_row="compat",
+        engine="python",
+        engine_kwargs=None,
         **kwargs,
     ):  # noqa: PR01, RT01, D200
         """
         Apply a function along an axis of the ``DataFrame``.
         """
-        if by_row != "compat":
+        if by_row != "compat" or engine != "python":
```

Comment:
```suggestion
        if by_row != "compat" or engine != "python" and engine_kwargs:
```
Should we update this condition to also take into account engine_kwargs?